src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -259,11 +254,18 @@ void maybeReassignPartitionAndElectLeader() throws Exception {
           reassignPartitions(zkUtils, brokers, _topic, partitionInfoList.size(), _replicationFactor);
         }
 
+        // Update the properties of the monitor topic if any config is different from the user-specified config
         Properties currentProperties = AdminUtils.fetchEntityConfig(zkUtils, ConfigType.Topic(), _topic);
-        if (!currentProperties.equals(_topicProperties)) {
+        Properties expectedProperties = new Properties();
+        for (Object key: currentProperties.keySet())
+          expectedProperties.put(key, currentProperties.get(key));
+        for (Object key: _topicProperties.keySet())
+          expectedProperties.put(key, _topicProperties.get(key));
+
+        if (!currentProperties.equals(expectedProperties)) {
```

Comment:
Kafka broker is configured with `min.insync.replicas`. If we don't explicitly specify optional properties such as `min.insync.replicas`, then the value in the broker config will be applied. The purpose of this patch is to allow kafka-monitor to use the default config if the Kafka cluster if possible.